### PR TITLE
resolve lotus, test-vectors, statediff dependency cycle

### LINKED ISSUE #3666: iron out the dependency cycles between lotus, test-vectors, and statediff
This is what we're planning to do:

- [x] test vector schema versioned as a separate go module (it will evolve very little), removing its dependencies on `abi` and `exitcode` packages, by replacing the types with their underlying (int64s and big.Int), and type converting on the driver.
    - This way we derisk the schema inadvertently forcing specific dependencies on lotus.
- [x] chaos actor moved into the Lotus codebase, and owned by the Oni team via CODEOWNERS.
    - This is how it should've been from day 1, but the chaos actor was in incubation and we needed to iterate on it rapidly. CODEOWNERS will solve the maintenance/PR approval problem.
    - With this, and the above checkbox, Lotus no longer will depend on test-vectors in the go.mod.
- [x] statediff not imported as a go package, but rather called as a CLI tool, if available in the host's path.
    - We built statediff as a standalone CLI tool that can take a CAR with two roots and perform the diff.
    - We will use that run mode of statediff, because it's also how other implementations (Forest and Fuhon) will call out to statediff.
    - This way we check two extra boxes: (1) we serve as a proper reference implementation without unfair advantage (being able to call statediff as a go module), and (2) we test the CLI mode of statediff.
- [ ] CODEOWNERS to streamline maintenance within the respective packages.

### PR DESCRIPTION
**Pull request by raulk**
Fixes #3666.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -231,6 +231,14 @@ jobs:
...
           gobin: $HOME/.local/bin
           version: 0.5.2
+      - run:
+          name: install statediff globally
+          command: |
+            mkdir -p /tmp/statediff
+            git clone https://github.com/filecoin-project/statediff.git /tmp/statediff
+            cd /tmp/statediff
+            go generate ./...
+            go install ./cmd/statediff
```
Comment (raulk) [inline .circleci/config.yml:241]: We now run statediff as a binary. This has two advantages:

1. allows us to drop the go dependency, and thus mitigate the risk of dependency contagion (e.g. if statediff uses a newer version of go-state-types than Lotus, minimal version selection would've inadvertently chosen the newer one 😨 ).
2. we use statediff just like other non-Go implementations would use it, which makes us even better as a reference implementation ;-)

File: conformance/runner_test.go [hunk 2]
```diff
@@ -256,6 +266,23 @@ func assertMsgResult(t *testing.T, expected *schema.Receipt, actual *vm.ApplyRet
...
+	// check if statediff exists; if not, skip.
+	if err := exec.Command("statediff", "--help").Run(); err != nil {
+		t.Log("could not dump 3-way state tree diff upon test failure: statediff command not found")
+		t.Log("install statediff with:")
+		t.Log("$ git clone https://github.com/filecoin-project/statediff.git")
+		t.Log("$ cd statediff")
+		t.Log("$ go generate ./...")
+		t.Log("$ go install ./cmd/statediff")
+		return
+	}
```
Comment (raulk) [inline conformance/runner_test.go:278]: If statediff is not available, we simply skip it and log instructions.

File: go.mod [hunk 3]
```diff
@@ -39,10 +40,10 @@ require (
 	github.com/filecoin-project/go-storedcounter v0.0.0-20200421200003-1c99c62e8a5b
 	github.com/filecoin-project/specs-actors v0.9.7
 	github.com/filecoin-project/specs-storage v0.1.1-0.20200907031224-ed2e5cd13796
-	github.com/filecoin-project/statediff v0.0.1
-	github.com/filecoin-project/test-vectors v0.0.0-20200907193218-2c0739eccce6
```
Comment (raulk) [inline go.mod:43]: 👋

File: .circleci/config.yml (hunk 1 above)
Comment (willscott) [reply .circleci/config.yml:238]: https://github.com/filecoin-project/statediff/pull/15 will allow removal of the `go generate`

**Review by willscott — APPROVED** [review]

**Review by alanshaw — APPROVED** [review]

**Review by magik6k — APPROVED** [review]

**Pull request merged**